spec/index.bs

```diff
@@ -2101,6 +2101,88 @@ For various image sources of {{GPUImageCopyExternalImage}}:
 
 Note: Check browser implementation support for these features before relying on them.
 
+## Numeric conversions from JavaScript to WGSL ## {#conversions-to-wgsl}
+
+Several parts of the WebGPU API ([=pipeline-overridable=] {{GPUProgrammableStage/constants}} and
+render pass clear values) take numeric values from WebIDL ({{double}} or {{float}}) and convert
+them to WGSL values (`bool`, `i32`, `u32`, `f32`, `f16`).
+
+<div algorithm>
+    To convert an IDL value |idlValue| of type {{double}} or {{float}} <dfn abstract-op>to WGSL type</dfn> |T|,
+    possibly throwing a {{TypeError}}:
+
+    1. [=Assert=] |idlValue| is a finite value, since it is not {{unrestricted double}} or {{unrestricted float}}.
+
+    1. Let |v| be the result of [=!=] converting |idlValue| to [=converted to an ECMAScript value|an ECMAScript value=].
+        <!-- This back-conversion is just here so we can call back into the ES->IDL conversion definitions from WebIDL. -->
+
+    1. <dl class=switch>
+            : If |T| is `bool`
+            ::
+                Return the WGSL `bool` value corresponding to the result of [=!=] converting |v| to
+                [=converted to an IDL value|an IDL value=] of type {{boolean}}.
+
+                Note: Because of the intermediate conversion through {{double}} or {{float}},
+                non-numeric, non-boolean values like `[]` and `{}` convert differently than they
+                would directly to IDL {{boolean}}.
```

Comment:
If the algorithm only operates on floats or doubles, when would this be relevant?

Reply:
The input to this algorithm is the "intermediate" double/float which comes from an IDL double/float. But in JavaScript you could pass a JS value, which gets converted to an IDL double/float, which then gets converted to a WGSL boolean. If the JS value was a boolean or number, the result is as expected, but if you pass other garbage then the results are weird.